src/openms_gui/source/VISUAL/DIALOGS/ToolsDialog.cpp

```diff
@@ -420,6 +434,24 @@ namespace OpenMS
     }
   }
 
+  void ToolsDialog::addEntries_()
+  {
+    QStringList names;
+    for (const auto& p : param_future_.results())
+    {
+      // Extract the tool's name
+      String name = p.begin().getName().substr(0, p.begin().getName().rfind(":"));
```

Comment:
does this still work if the INI file was not read correctly? Then the param object is empty, right?
Also, you only need that if the layer_type_ matches. So evaluation can be deferred to the actual push_back into `names`?